File: frontend/src/types/task.ts
export type Priority = "low" | "medium" | "high";
export type TaskState = "PENDING" | "EXECUTING" | "DONE";

export interface TaskItem {
  id: number;
  title: string;
  description?: string;
  completed: boolean;
  dueDate: string | null;
  priority: Priority;
  state: TaskState;
  createdAt: string;
}


export interface Subtask {
  id: number;
  title: string;
  completed: boolean;
  taskId: number;
}

export interface EditData {
  title: string;
  dueDate: string;
  priority: Priority;
}

File: backend/src/main/java/com/example/taskapp/entity/Task.java
package com.example.taskapp.entity;

import jakarta.persistence.*;
import lombok.Data;
import java.time.LocalDateTime;
import java.time.LocalDate;

@Data
@Entity
@Table(name = "tasks")
public class Task {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String title;           // タスク名
    private String description;     // 詳細
    private boolean completed;      // 完了フラグ
    private LocalDateTime createdAt; // 作成日時

    private LocalDate dueDate;   // 締切日
    private String priority;     // 優先度（low, medium, high）

    // 🔥 ユーザーと紐づける（ここが重要！）
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id")
    private User user;
}
